CMakeLists.txt

```diff
@@ -79,6 +79,8 @@ add_subdirectory(include)
 add_subdirectory(lib)
 add_subdirectory(tools)
 add_subdirectory(test)
+add_subdirectory(sycl_runtime)
```

Comment:
Great to see the PR!
I would suggest following MLIR upstream for placing the runtimes.
(https://github.com/llvm/llvm-project/tree/main/mlir/lib/ExecutionEngine)
Upstream puts code for shared libraries called from LLVM generated code under "lib/ExecutionEngine"